### PR TITLE
♻️Autoscaling: prepare for computational autoscaling

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
- [x] refactoring of code to prepare for computational cluster autoscaling

## Related issue/s
- pre-requisite of https://github.com/ITISFoundation/osparc-simcore/pull/4711

### CODE REVIEW

File: services/autoscaling/src/simcore_service_autoscaling/modules/auto_scaling_base.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    assert app_settings.AUTOSCALING_EC2_INSTANCES  # nosec
+
+    # get current docker nodes (these are associated (active or drained) or disconnected)
+    docker_nodes: list[Node] = await utils_docker.get_monitored_nodes(
+        get_docker_client(app),
+        node_labels=app_settings.AUTOSCALING_NODES_MONITORING.NODES_MONITORING_NODE_LABELS,
+    )
+
+    # get the EC2 instances we have
+    existing_ec2_instances = await get_ec2_client(app).get_instances(
```
Comment (pcrespov) [inline services/autoscaling/src/simcore_service_autoscaling/modules/auto_scaling_base.py:39]: MINOR: what is an instance? annotate for clarity?

File: services/autoscaling/src/simcore_service_autoscaling/modules/auto_scaling_base.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                new_found_instances.append(AssociatedInstance(new_node, instance_data))
+            else:
+                still_pending_ec2s.append(instance_data)
+        except Ec2InvalidDnsNameError:
+            _logger.exception("Unexpected EC2 private dns")
+    # NOTE: first provision the reserve drained nodes if possible
+    all_drained_nodes = (
+        cluster.drained_nodes + cluster.reserve_drained_nodes + new_found_instances
+    )
+    return dataclasses.replace(
```
Comment (pcrespov) [inline services/autoscaling/src/simcore_service_autoscaling/modules/auto_scaling_base.py:125]: this is an interesting one ...

**Review by pcrespov — APPROVED** [review]
spotless! :tada:

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

**Pull request merged**